Updates reading through the code and commits.

benh opened the pull request.

benh committed:
Used 'flags.docker' in tests.

And some other minor style cleanups.

benh committed:
Exposed and used the Docker container name prefix.

benh committed:
Renamed variable to be more accurate.

Since we actually have a Docker::Container abstraction I
s/dockerContainer/dockerContainerizer/.

benh committed:
Refactored Docker::Container::pid() to return an Option.

Safer to check for 'None' than 0. Also fixed a compilation bug
comparing signed vs unsigned values with CHECK_NE.

benh committed:
Made Docker::ps/rm default arguments match CLI defaults.

tnachen commented:
All the changes looks good! Thanks for the clean up and style changes again, still not totally remembering all the rules. Perhaps I will also help out on style check as it's hard for new contributors to ttoally grok these in mind.

benh committed:
Be more mock friendly in MockDockerContainerizer.

Rather than having the MockDockerContainerizer store local variables
when certain events occur we can make the tests store what ever things
they need by mocking out the methods.

benh commented:
Committed via command line.
